service/src/main/java/greencity/service/UserNotificationServiceImpl.java

```diff
@@ -447,6 +521,59 @@ private void generateNotification(final UserVO targetUserVO, final UserVO action
         sendNotification(notification.getTargetUser().getId());
     }
 
+    private CompletableFuture<PageableAdvancedDto<NotificationDto>> getNotificationsForUserFromGreenCity(
+        ExecutorService executorService, Pageable page,
+        Principal principal, String language, ProjectName projectName, List<NotificationType> notificationTypes,
+        Boolean viewed) {
+        CompletableFuture<PageableAdvancedDto<NotificationDto>> completableFuture = new CompletableFuture<>();
+        completableFuture.completeAsync(
```

Comment:
Can we use `CompletableFuture.supplyAsync` instead of creating` new CompletableFuture()` for every method call?